src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -483,6 +484,73 @@ private IOperation CreateBoundUnconvertedAddressOfOperatorOperation(BoundUnconve
                 boundUnconvertedAddressOf.WasCompilerGenerated);
         }
 
+        private bool TryGetAttributeData(BoundAttribute boundAttribute, [NotNullWhen(returnValue: true)] out AttributeData? data)
+        {
+            if (boundAttribute.HasAnyErrors ||
+                boundAttribute.Syntax.Parent?.Parent is not { } attributedDeclaration)
+            {
+                data = null;
+                return false;
+            }
+
+            var model = _semanticModel.Compilation.GetSemanticModel(boundAttribute.SyntaxTree!);
```

Comment:
Can't use the original `_semanticModel` here. It is an `AttributeSemanticModel` (derives from `MemberSemanticModel`). It always returns `null` from `GetDeclaredSymbol` overloads:

https://github.com/dotnet/roslyn/blob/e103096bfdb5a7652abd4d2d7dcbc855354366a8/src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.cs#L589-L593